demo/admin/src/products/future/generated/ProductForm.tsx

```diff
@@ -75,26 +75,36 @@ type FormValues = Omit<ProductFormDetailsFragment, "dimensions"> & {
     };
     image: BlockState<typeof rootBlocks.image>;
 };
+type InitialFormValues = {
+    title?: string;
+    dimensions?: {
+        width?: string;
+    };
+    inStock?: boolean;
+    image?: BlockState<typeof rootBlocks.image>;
+};
 
 interface FormProps {
+    width?: number;
+    title?: string;
     id?: string;
 }
 
-export function ProductForm({ id }: FormProps): React.ReactElement {
+export function ProductForm({ width, title, id }: FormProps): React.ReactElement {
     const client = useApolloClient();
     const mode = id ? "edit" : "add";
-    const formApiRef = useFormApiRef<FormValues>();
+    const formApiRef = useFormApiRef<FormValues, InitialFormValues>();
     const stackSwitchApi = useStackSwitchApi();
 
     const { data, error, loading, refetch } = useQuery<GQLProductQuery, GQLProductQueryVariables>(
         productQuery,
         id ? { variables: { id } } : { skip: true },
     );
 
-    const initialValues = React.useMemo<Partial<FormValues>>(
+    const initialValues = React.useMemo<Partial<FormValues> | InitialFormValues>( // Partial<FormValues> must be compatible with InitialFormValues (e.g. changing inStock: boolean to inStock?:boolean) because <FinalForm initialValues only accepts InitialFormValues. Maybe use "Partial<FormValues> | InitialFormValues" as second generic argument
```

Comment:
This should only be `React.useMemo<InitialFormValues>`

Reply:
I tried your changes from demo and removed satisfies but this does not prevent setting random fields. is this alright? I added loremIpsum and loremIpsumDolor fields, but typescript does not complain. But if I assign a number to title it's detected as problem.

<img width="1016" alt="image" src="https://github.com/user-attachments/assets/f527b836-eecc-46aa-a694-fb19714b9a4b">

Reply:
> I tried your changes from demo and removed satisfies but this does not prevent setting random fields. is this alright?

This is due to the nature how TypeScript handles function return types (which useMemo is). A function can return more than the specified type. So returning `{ foo: "bar", bar: "foo" }` for a function `() => { foo: string }` is perfectly fine.

If you really want to enforce the type, you'd need to create a variable (e.g, `initialValues`) and return it:

```tsx
const initialValues: InitialFormValues = {
    image: rootBlocks.image.defaultValues(),
    inStock: false,
    additionalTypes: [],
    tags: [],
    dimensions: { width },
    loremIpsumDolor: 123,
};
return initialValues;
```

But I'd neglect that. You could use satisifes for that, but I'm not sure if we need/want it. Also, this should be a separate PR, since it would be a general decision, I suppose.